cli/src/text_util.rs

```diff
@@ -35,6 +37,33 @@ pub fn split_email(email: &str) -> (&str, Option<&str>) {
     }
 }
 
+pub fn render_copied_path(source: &str, target: &str) -> String {
+    if source == target {
+        return target.into();
+    }
+    let prefix = UnicodeSegmentation::split_word_bound_indices(source)
+        .zip(UnicodeSegmentation::split_word_bound_indices(target))
+        .take_while_inclusive(|((_, s), (_, t))| s == t)
+        .last()
+        .map(|((s, _), (t, _))| (s, t))
+        .unwrap_or((0, 0));
+    let suffix = UnicodeSegmentation::split_word_bound_indices(source)
+        .rev()
+        .zip(UnicodeSegmentation::split_word_bound_indices(target).rev())
+        .take_while(|((_, s), (_, t))| s == t)
+        .last()
+        .map(|((s, _), (t, _))| (s, t))
+        .unwrap_or((source.len(), target.len()));
```

Comment:
Nit: the whole block needs readable names, iiuc `s` => `source`, `t` => `target`

Reply:
done